### PR TITLE
[Wait for #2536][application] add generate_multiple_tokens for llm @open sesame 04/05 16:14

### PR DESCRIPTION
**Pull request by baek2sm**
Added generate_multiple_tokens function for first generation on llm.

This function takes one logits and generates multiple output tokens.
To meet the purpose of the target application,
even if input are multiple logits,
only the first logits is used to generate multiple output tokens.

**Self evaluation:**
1. Build test:	 [X]Passed [ ]Failed [ ]Skipped
2. Run test:	 [X]Passed [ ]Failed [ ]Skipped

### CODE REVIEW

**Review by jijoongmoon — APPROVED** [review]
LGTM

**Review by dkjung — COMMENTED** [review]
LGTM

**Review by dkjung — APPROVED** [review]
LGTM

**Review by jijoongmoon — APPROVED** [review]
LGTM

**Review by skykongkong8 — APPROVED** [review]
lgtm

**Review by SeoHyungjun — APPROVED** [review]
LGTM

**Review by djeong20 — APPROVED** [review]
Nice work! please rebase to resolve the conflict.

File: Applications/LLaMA/jni/main.cpp [hunk 1]
```diff
@@ -219,6 +219,39 @@ std::vector<int> generate(float *logits, unsigned int NUM_VOCAB = 0,
...
 }
 
+std::vector<int> generate_multi_tokens(
+  float *logits, unsigned int NUM_VOCAB = 0, unsigned int NUM_TARGET_TOKENS = 1,
+  float repetition_penalty = 1, unsigned int *input_ids = nullptr,
+  unsigned int NUM_INPUT_IDS = 0, unsigned int *bad_words_ids = nullptr,
+  unsigned int NUM_BAD_WORDS_IDS = 0) {
+  std::vector<int> outputs;
+  // apply repetition penalty
+  if (repetition_penalty != 1 && input_ids != nullptr && NUM_INPUT_IDS != 0) {
```
Comment (lhs8928) [inline Applications/LLaMA/jni/main.cpp:229]: repetition_penalty is a float value so instead of using != operator please do something like this
`abs(repetition_penalty - 1) < eps`

**Review by lhs8928 — CHANGES_REQUESTED** [review]

File: Applications/LLaMA/jni/main.cpp (hunk 1 above)
Comment (baek2sm) [reply Applications/LLaMA/jni/main.cpp]: Thanks, I applied it.

**Review by lhs8928 — APPROVED** [review]
LGTM
